app/brave_main_delegate_unittest.cc

```diff
@@ -46,16 +46,9 @@ TEST(BraveMainDelegateUnitTest, OverrideSwitchFromCommandLine) {
   base::CommandLine& command_line = *base::CommandLine::ForCurrentProcess();
```

Comment:
Frankly, the relationship between there tests and `VariationsCommandLineUtils` isn't clear to me.

We still directly checking `kVariationsServerURL` the `BraveMainDelegateUnitTest.DefaultCommandLineOverrides` to be sure that the function is called (not using a mock calls)

`VariationsCommandLineUtils.DefaultVariationsServerUrl` checks the same thing.

Reply:
think of it as "integration" test for all command line overrides we do in `BraveMainDelegate`. They overlap a bit, but I think it's fine.